drivers/source/SFDP.cpp

```diff
@@ -134,5 +174,127 @@ int sfdp_parse_headers(Callback<int(bd_addr_t, void *, bd_size_t)> sfdp_reader,
+}
+
+size_t sfdp_detect_page_size(uint8_t *basic_param_table_ptr, size_t basic_param_table_size)
+{
+    constexpr int SFDP_BASIC_PARAM_TABLE_PAGE_SIZE = 40;
+    constexpr int SFDP_DEFAULT_PAGE_SIZE = 256;
+
+    unsigned int page_size = SFDP_DEFAULT_PAGE_SIZE;
+
+    if (basic_param_table_size > SFDP_BASIC_PARAM_TABLE_PAGE_SIZE) {
+        // Page Size is specified by 4 Bits (N), calculated by 2^N
+        int page_to_power_size = ((int)basic_param_table_ptr[SFDP_BASIC_PARAM_TABLE_PAGE_SIZE]) >> 4;
+        page_size = 1 << page_to_power_size;
+        tr_debug("Detected Page Size: %d", page_size);
+    } else {
+        tr_debug("Using Default Page Size: %d", page_size);
+    }
+    return page_size;
+}
+
+int sfdp_detect_erase_types_inst_and_size(uint8_t *bptbl_ptr, sfdp_hdr_info &sfdp_info)
+{
+    uint8_t bitfield = 0x01;
+
+    // Erase 4K Inst is taken either from param table legacy 4K erase or superseded by erase Instruction for type of size 4K
+    if (sfdp_info.bptbl.size > SFDP_BASIC_PARAM_TABLE_ERASE_TYPE_1_SIZE_BYTE) {
+        // Loop Erase Types 1-4
+        for (int i_ind = 0; i_ind < 4; i_ind++) {
+            sfdp_info.smptbl.erase_type_inst_arr[i_ind] = -1; // Default for unsupported type
+            sfdp_info.smptbl.erase_type_size_arr[i_ind] = 1
+                                                          << bptbl_ptr[SFDP_BASIC_PARAM_TABLE_ERASE_TYPE_1_SIZE_BYTE + 2 * i_ind]; // Size is 2^N where N is the table value
+            tr_debug("Erase Type(A) %d - Inst: 0x%xh, Size: %d", (i_ind + 1), sfdp_info.smptbl.erase_type_inst_arr[i_ind],
+                     sfdp_info.smptbl.erase_type_size_arr[i_ind]);
+            if (sfdp_info.smptbl.erase_type_size_arr[i_ind] > 1) {
+                // if size==1 type is not supported
+                sfdp_info.smptbl.erase_type_inst_arr[i_ind] = bptbl_ptr[SFDP_BASIC_PARAM_TABLE_ERASE_TYPE_1_BYTE
+                                                                        + 2 * i_ind];
+
+                if ((sfdp_info.smptbl.erase_type_size_arr[i_ind] < sfdp_info.smptbl.regions_min_common_erase_size)
+                        || (sfdp_info.smptbl.regions_min_common_erase_size == 0)) {
+                    //Set default minimal common erase for signal region
+                    sfdp_info.smptbl.regions_min_common_erase_size = sfdp_info.smptbl.erase_type_size_arr[i_ind];
+                }
+                sfdp_info.smptbl.region_erase_types_bitfld[0] |= bitfield; // If there's no region map, set region "0" types bitfield as default
+            }
+
+            tr_debug("Erase Type %d - Inst: 0x%xh, Size: %d", (i_ind + 1), sfdp_info.smptbl.erase_type_inst_arr[i_ind],
+                     sfdp_info.smptbl.erase_type_size_arr[i_ind]);
+            bitfield = bitfield << 1;
+        }
+    } else {
+        tr_debug("SFDP erase types are not available - falling back to legacy 4k erase instruction");
+
+        // 0xFF indicates that the legacy 4k erase instruction is not supported
```

Comment:
[minor] Perhaps we could add an `SFDP_ERASE_BITMASK_TYPE_4K_ERASE_UNSUPPORTED` to replace the `0xff` below and get rid of this comment?

Reply:
Fixed.